qiskit/circuit/controlflow/builder.py

```diff
@@ -36,6 +38,125 @@
     import qiskit
 
 
+class CircuitScopeInterface(abc.ABC):
+    """An interface that circuits and builder blocks explicitly fulfil, which contains the primitive
+    methods of circuit construction and object validation.
+
+    This allows core circuit methods to be applied to the currently open builder scope, and allows
+    the builders to hook into all places where circuit resources might be used.  This allows the
+    builders to track the resources being used, without getting in the way of
+    :class:`.QuantumCircuit` doing its own thing.
+
+    This is a Qiskit-internal interface."""
```

Comment:
I removed the redundant line in 34b47dd44a, to avoid giving the impression that the statement has meaning - we want people to know that the stability policy is the reason.